src/openforms/submissions/utils.py

```diff
@@ -84,15 +86,50 @@ def send_confirmation_email(submission: Submission):
         logevent.confirmation_email_skip(submission)
         return
 
-    html_content = email_template.render(submission)
-    text_content = email_template.render(submission, {"rendering_text": True})
+    if (
+        submission.form.confirmation_email_option
+        == ConfirmationEmailOptions.form_specific_email
+    ):
+        try:
+            email_template = submission.form.confirmation_email_template
+            subject = email_template.subject
+            html_content = email_template.render(submission)
+            text_content = email_template.render(submission, {"rendering_text": True})
+        except (AttributeError, ObjectDoesNotExist):
```

Comment:
All this should not be needed/possible if the validation is set up properly.

Please validate on the form serializer that `ConfirmationEmailOptions.form_specific_email` can only be picked if both subject/content of form-specific templates are present.